mesonbuild/modules/rust.py

```diff
@@ -156,12 +156,14 @@ def test(self, state: ModuleState, args: T.Tuple[str, BuildTarget], kwargs: Func
         new_target_kwargs = base_target.original_kwargs.copy()
         # Don't mutate the shallow copied list, instead replace it with a new
         # one
-        new_target_kwargs['rust_args'] = \
-            new_target_kwargs.get('rust_args', []) + kwargs['rust_args'] + ['--test']
         new_target_kwargs['install'] = False
         new_target_kwargs['dependencies'] = new_target_kwargs.get('dependencies', []) + kwargs['dependencies']
         new_target_kwargs['link_with'] = new_target_kwargs.get('link_with', []) + kwargs['link_with']
 
+        lang_args = copy.copy(base_target.extra_args)
```

Comment:
`copy.copy` is a non-recursive copy. Is it what you want? Or do you want `copy.deepcopy`?

Reply:
I think that's correct, we only modify the dict container. I usually prefer `base_target.extra_args.copy()` syntax in this case, but that's up to you.

Reply:
I looked at this hard and I think using a shallow copy is okay here. I did remove the use of the `copy` module, and just used `dict.copy()`, since that's what the rest of the code does anyway